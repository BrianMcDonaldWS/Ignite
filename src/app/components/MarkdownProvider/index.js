import React from 'react';
import ReactRouterPropTypes from 'react-router-prop-types';

import App from '../App';

const noDocsFound = () => (
  <div>Hmmmm, somethings wrong. No docs files found....</div>
);

let internalUpdateCallback = () => {};

export const update = (...args) => internalUpdateCallback(...args);

export default class MarkdownProvider extends React.Component {
  static propTypes = {
    location: ReactRouterPropTypes.location.isRequired
  };

  constructor(props) {
    super(props);

    this.state = {
      markdown: {
        docRootIndexFile: noDocsFound
      }
    };

    internalUpdateCallback = this.onUpdate;
  }

  onUpdate = (path, component, isIndex, firstLink) => {
    const { markdown } = this.state;

    markdown[path] = component;

    if (isIndex) {
      markdown.docRootIndexFile = component;
      markdown.firstPagePath = firstLink;
    }

    this.setState({
      markdown
    });
    return component;
  };

  render() {
    return (
      <App markdown={this.state.markdown} location={this.props.location} />
    );
  }
}
