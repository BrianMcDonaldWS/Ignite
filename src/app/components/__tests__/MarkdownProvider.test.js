import React from 'react';

import MarkdownProvider, { update } from '../MarkdownProvider';
import renderToJson from './util';

test('Has default route', () => {
  expect(renderToJson(<MarkdownProvider />)).toMatchSnapshot();
});

test('update markdown provided', () => {
  update('/foo/bar', () => <h1> Simple Component </h1>);
  expect(renderToJson(<MarkdownProvider />)).toMatchSnapshot();
});

test('sets first page and root from index file', () => {
  update('/foo/bar', () => <h1> Simple Component </h1>, true, '/first/page');
  expect(renderToJson(<MarkdownProvider />)).toMatchSnapshot();
});
