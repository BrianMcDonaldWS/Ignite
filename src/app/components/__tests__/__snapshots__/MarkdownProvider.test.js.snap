// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has default route 1`] = `
<App
  location={undefined}
  markdown={
    Object {
      "docRootIndexFile": [Function],
    }
  }
/>
`;

exports[`sets first page and root from index file 1`] = `
<App
  location={undefined}
  markdown={
    Object {
      "/foo/bar": [Function],
      "docRootIndexFile": [Function],
      "firstPagePath": "/first/page",
    }
  }
/>
`;

exports[`update markdown provided 1`] = `
<App
  location={undefined}
  markdown={
    Object {
      "/foo/bar": [Function],
      "docRootIndexFile": [Function],
    }
  }
/>
`;
